### PR TITLE
list-view,big-list-view and fixed-size-list

### PR DESCRIPTION
**Pull request by DerThorsten**
generalized the list layout impl st. all list like layouts (list, list-view, fixed-sized-list) are all implemented with the same core implementation.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -79,7 +79,9 @@ jobs:
 
     - name: Run tests
       working-directory: build
-      run: cmake --build . --config ${{matrix.config.name}} --target run_tests_with_junit_report
+      run:    ./bin/${{matrix.config.name}}/test_sparrow_lib  # cmake --build . --config ${{matrix.config.name}} --target run_tests_with_junit_report
```
Comment (Alex-PLACET) [inline .github/workflows/linux.yml:82]: If you don't use the cmake target, the report is not generated

File: include/sparrow/layout/list_layout/list_array.hpp [hunk 2]
```diff
@@ -96,78 +148,208 @@ namespace sparrow
         inner_const_reference value(size_type i) const;
         
         // data members
-        flat_array_offset_type * p_list_offsets;
+        //flat_array_offset_type * p_list_offsets;
```
Comment (Alex-PLACET) [inline include/sparrow/layout/list_layout/list_array.hpp:151]: To remove

File: include/sparrow/layout/list_layout/list_array.hpp [hunk 3]
```diff
@@ -96,78 +148,208 @@ namespace sparrow
...
+    
+    inline fixed_sized_list_array::fixed_sized_list_array(arrow_proxy proxy)
+    :   base_type(std::move(proxy)),
+        m_list_size(0)
+    {
+        // get the list size from the format string
+        const auto format = std::string(this->storage().format());
+        const auto n_digits = format.size() - 3;
+        const auto list_size_str = format.substr(3, n_digits);
+        m_list_size = std::stoull(list_size_str);
```
Comment (Alex-PLACET) [inline include/sparrow/layout/list_layout/list_array.hpp:348]: Should be extracted in a function which will be used in l.342.
Also it seems that some checks are missing like format.size() >= 3

Comment (DerThorsten) [reply include/sparrow/layout/list_layout/list_array.hpp:348]: with an assert in debug more or a real runtime check?

Comment (Alex-PLACET) [reply include/sparrow/layout/list_layout/list_array.hpp:348]: I would say an assert. But I'm thinking if it should be checked in the proxy. As a constructed proxy should be a valid arrow array + schema.

**Review by Alex-PLACET — APPROVED** [review]
LGTM, if you can, please run the clang_format cmake target (you have to define -DACTIVATE_LINTER=ON first)

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**